### PR TITLE
chore(endpoint): add codegen changes for endpoint ruleset compression

### PR DESCRIPTION
**Pull request by kuhe**
This is a follow-up to https://github.com/aws/aws-sdk-js-v3/pull/4300 that adds the actual codegen changes.

It integrates the compression step into the client generation script and prevents prettier from formatting the ruleset files.

- [x] `yarn generate-clients`
- [x] `yarn build:all`
- [x] `(cd tests/endpoints-2.0 && npx jest)`
- [x] `yarn test:integration:legacy`

### CODE REVIEW

File: clients/client-accessanalyzer/src/endpoint/ruleset.ts [hunk 1]
```diff
@@ -1,320 +1,31 @@
...
+export const ruleSet: RuleSetObject = _data as any;
```
Comment (trivikr) [inline clients/client-accessanalyzer/src/endpoint/ruleset.ts:31]: Generated clients need to be updated after updates in https://github.com/aws/aws-sdk-js-v3/pull/4300
```suggestion
export const ruleSet: RuleSetObject = _data;
```

File: .prettierignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ CHANGELOG.md
 .github/*
 **/*.hbs
 **/*/report.md
+**/*/ruleset.ts
```
Comment (trivikr) [inline .prettierignore:6]: Be explicit

```suggestion
clients/*/src/endpoint/ruleset.ts
```

**Review by trivikr — CHANGES_REQUESTED** [review]

File: clients/client-accessanalyzer/src/endpoint/ruleset.ts (hunk 1 above)
Comment (kuhe) [reply clients/client-accessanalyzer/src/endpoint/ruleset.ts]: just finished rerunning codegen

File: .prettierignore (hunk 2 above)
Comment (trivikr) [inline .prettierignore:6]: Looks like force push overrode the changes?
```suggestion
clients/*/src/endpoint/ruleset.ts
```

**Review by trivikr — APPROVED** [review]

**Pull request merged**